src/beatree/mod.rs

```diff
@@ -110,31 +112,27 @@ impl Tree {
             staged_changeset,
             root,
             &mut branch_node_pool,
-            &mut leaf_store_tx,
+            &mut sync.leaf_store_wr,
         )
         .unwrap();
 
-        let (ln, ln_freelist_pn, ln_bump, ln_extend_file_sz) = {
-            let o = leaf_store_tx.commit();
-            let ln: Vec<()> = o
-                .to_allocate
-                .into_iter()
-                .chain(o.exceeded)
-                .map(|(_pn, _page)| ())
-                .collect();
-            let ln_freelist_pn = o.new_free_list_head.0;
-            let ln_bump = 0; // TODO: commit should return the bump;
-            let ln_extend_file_sz: Option<u64> = None;
-            (ln, ln_freelist_pn, ln_bump, ln_extend_file_sz)
+        let (ln, freelist_head_pn, ln_bump, ln_extend_file_sz) = {
+            let LeafStoreCommitOutput {
+                pages,
+                extend_file_sz,
+                freelist_head_pn,
+                bump,
+            } = sync.leaf_store_wr.commit();
+            (pages, freelist_head_pn, bump, extend_file_sz)
         };
 
         // TODO: BBN dumping.
 
         let new_meta = Meta {
             sync_seqn,
             next_bbn_seqn,
-            ln_freelist_pn,
-            ln_bump,
+            ln_freelist_pn: freelist_head_pn.0,
+            ln_bump: ln_bump.0,
```

Comment:
Nit: tbh doesn't feel like an improvement compared to the prev version:

1. pesky .0 are not hidden now
2. we don't use the short-hand init syntax
3. the name `ln_freelist_pn` woulda worked better, esp. in the context of recent changes of our thinking, that there may appear the second freelist.

(once again, this is purely stylistic)

Reply:
Agreed! I moved the `.0` to where `sync.leaf_store_wr.commit` is called, making it much cleaner now!